### ISSUE TITLE
Cannot use Ambient mode with AW3423DW 

### ISSUE THREAD

**Issue by MaluNoPeleke** [T-Troll/alienfx-tools#222]
**Describe the bug**
As of release 7.2.2 and accordingly to #194 the Dell Alienware monitor AW3423DW is supported. I just discovered alienfx-tools and want to use it to synchronize the RGBs of this monitor with the screen content (called Ambient mode I guess).
After launching the application and detecting the monitor I set it to "Ambient" but the RGBs just slowly rotate colors, even with a static screen.

**To Reproduce**
1. Start alienfx-gui
2. Click on 'Effects mode dropdown' and set it to "Ambient"
3. Click on "Refresh"

**Expected behavior**
The RGB should mirror the current screen content color.

**Screenshots**
![image](https://user-images.githubusercontent.com/1446541/196518833-fca00fa1-ca1b-448f-8f8e-3bec1ac98fde.png)

**System (please complete the following information):**
 - Model: AlienWare AW3423DW
 - OS: Windows 11 insider

**Additional context**
I cannot select any other tab in the window such as "Colors", "Ambient" or "Haptics". Only "Devices and Grids" is selected/visible by default.

**Comment by T-Troll** [comment]
>  but the RGBs just slowly rotate colors, even with a static screen

This happened because you need to set up Ambient - screen zones not defined yet. So colors not changed, using you default settings.

> I cannot select any other tab in the window such as "Colors", "Ambient" or "Haptics". Only "Devices and Grids" is selected/visible by default.

Aha. This is the bug. Let me check, seems like some data is not set correctly after detection. For now, just try to restart app - dit it work correctly after this?

**Comment by T-Troll** [comment]
Yes, it was a bug after "Detect" button pressed and data applied.
Fixed in 7.4.3.1.

**Comment by MaluNoPeleke** [comment]
Thanks for the quick fix, now I was indeed able to switch to the other areas but after defining ambient zones and clicking on "Refresh" the LEDs still change colors slowly as before.
Anything else I missed?
![image](https://user-images.githubusercontent.com/1446541/196626281-b597ef38-7dbc-4330-81d4-86d6fcf02b1c.png)

**Comment by T-Troll** [comment]
Do you have DirectX installed? (install if not yet)
Did the colors of "screen zones" represent real screen colors correctly? (press "Reset"/choose other screen if you have more, then one if not).
BTW, read tools readme - there are some hints why it can not work (in "requirements" and "issues").

I'll check it at my system, all correct.

**Comment by MaluNoPeleke** [comment]
Yes, I have DirectX installed and the colors of the screen zones represent real screen colors correctly.
Here is a video of the issue: https://photos.app.goo.gl/qGJdf8Vc6mNLswUv9

**Comment by T-Troll** [comment]
Got it.

Seems like something/someone took control about lights. If it was RGB keyboard, i've said global effect active.
Can you please check just plain color set? Did it work?
If not, i need to check you AWCC settings.

PS: Also, it's a good idea to stop AWCCService/chek "Disable AWCC" at Settings. But, seems like the issue is deeper a bit...

**Comment by MaluNoPeleke** [comment]
I have checked that option already but that doesn't help.

**Comment by MaluNoPeleke** [comment]
Changing Colors doesn't do anything either: 
![image](https://user-images.githubusercontent.com/1446541/196800227-e2e45f08-67b8-45d5-b25f-6b552f80971d.png)
These are my other settings:
![image](https://user-images.githubusercontent.com/1446541/196800444-e6b2d285-7b9b-4cf9-8748-8f562aa47fd9.png)
![image](https://user-images.githubusercontent.com/1446541/196800497-c787499c-4eb3-43bf-8503-c5aa5396e533.png)
If I run AWCC it looks like this (didn't set up anything there yet):
![image](https://user-images.githubusercontent.com/1446541/196800981-77a7a7b6-11eb-48e3-994d-0d890fa1b728.png)
![image](https://user-images.githubusercontent.com/1446541/196801024-4d377992-507c-48b2-a886-ae305ac34731.png)
If I change colors there it works as expected immediately.

**Comment by T-Troll** [comment]
Good, so it's not a hardware issue, just configuration. I may break it inside tools, or it's a result of incorrect initialization (we spend a lot of time to do it right way then add support).

I need to check what do you have into "FX" settings of the AWCC (choose monitor device first), fortunately i can read German.
Also, it was a post at Alienware reddit - this can be monitor menu issue, latest firmware have some bugs there.

Anyway, we need to check what happened on hardware level later on, so please install USBPcap - i'll provide you test version and tell what to do.

**Comment by MaluNoPeleke** [comment]
This is how the FX area looks like (still untouched):
![image](https://user-images.githubusercontent.com/1446541/196863786-1c113140-e11f-4a1e-803e-6d6916ba6dc9.png)
![image](https://user-images.githubusercontent.com/1446541/196863689-10a24cb9-a126-4b15-9915-01a539c44ef0.png)

**Comment by T-Troll** [comment]
Thank you.

What is interesting here - you set lights to color, but backlights are breathing! Please check back panel lights settings (monitor back picture at up-right).

Anyway, seems like color set (or device reset again!) is broken. Can you, please:
1. Close AWCC
2. Open Services
3. Stop AWCCService
4. Run some CLI commands, like `alienfx-cli set-all=255,0,0` or `alienfx-cli set-all=0,0,255`

Did it change monitor light's color?

**Comment by MaluNoPeleke** [comment]
This is the output:

> C:\Program Files\AlienFX Tools>alienfx-cli set-all=0,0,255
> alienfx-cli v7.4.3.1
> Dell API not found, 1 low-level devices found.
> Done.

The color always changes to purple for a second, no matter what kind of RGB combination I use.

**Comment by T-Troll** [comment]
Thank you, things starting to be clear.

So what we have? CLI change color (GUI share the same code, but more complex way), but something in system or monitor configuration reset it back immediately. And seems like it's not AWCC.

Unfortunately, i doesn't have this monitor at my table, so i can only think and guess what is the reason for this behavior...

... Interesting case! Let me think a bit. Meanwhile, try to Reset lighting and set "Effects" on, then check what happened. And, most probably, we need to check what happened at USB level - both from AWCC and my apps (USBPcap is for this).

**Comment by MaluNoPeleke** [comment]
This is how it looks like after reset: 
![PXL_20221020_084500990](https://user-images.githubusercontent.com/1446541/196901458-cf27167f-a284-4137-ba37-6afdd24365ac.jpg)
The effects setting provides three options: Spectrum, Rainbow Cycle and Comet.
USBPcap is installed, what to do with it?

**Comment by T-Troll** [comment]
Ok, let's try,
First, please get this CLI version - [alienfx-cli.zip](https://github.com/T-Troll/alienfx-tools/files/9843217/alienfx-cli.zip)
It has older device init command (it provides side effect - all lights will be white).
Stop AWCCService and check `set-all` command like before. Did it work now?

If not, we will grab your version of AWCC interface to check.

**Comment by MaluNoPeleke** [comment]
No, still same behavior as before, just with a slightly different output on cmd:

> alienfx-cli v7.4.3.1
> Dell API ready, 1 low-level devices found.
> Done.

Which file of AWCC do you need?

@translucentfocus does it work for you?

**Comment by T-Troll** [comment]
Thanks for information.
Uff... We need to do it a hard way.

Can you, please:
1. Install USBPcap (you are already do so AFAIR).
2. Run USBPcap
3. Check what USB hub have monitor connected (the list of devices will be print into it's console) and select it
4. Select any file name (f.e. `data.pcap`)
5. Keep it running
6. Open AWCC
7. Set some monitor lights (better use simple colors - red, green, blue, etc)
8. Stop USBPcap (switch to it's window and press CTRL+C)
9. Share file from p.4 here

I want to check how AWCC init it now, as well as possible interface changes.

**Comment by translucentfocus** [comment]
> No, still same behavior as before, just with a slightly different output on cmd:
> 
> > alienfx-cli v7.4.3.1
> > Dell API ready, 1 low-level devices found.
> > Done.
> 
> Which file of AWCC do you need?
> 
> @translucentfocus does it work for you?

I haven't tested on the sample we have... but I see you're using Windows 11... does Windows 11 use DX12 for the desktop compositor?  I know DX12 doesn't allow access to required information.  @T-Troll will probably know, but I just wasn't even sure Ambient mode _would_ work in Windows 11.

**Comment by T-Troll** [comment]
@translucentfocus, yes, you are right about DX12, i even mention it into readme. But it should work for desktop, at least.
But here we have another issue - light control simple broken at his device. We are trying to configure out why it happened.

JFYI: Here is the issue of monitors support story - https://github.com/T-Troll/alienfx-tools/issues/194

**Comment by MaluNoPeleke** [comment]
Here it is:
[data.zip](https://github.com/T-Troll/alienfx-tools/files/9863907/data.zip)

**Comment by T-Troll** [comment]
Sorry for delay.

It's a mystery. You have the same device ID as before, but it used a completely different interface to set lights!
But, being honest, more simple one.

I don't know, is it a firmware or different hardware mode.
Anyway, let me implement it.

**Comment by MaluNoPeleke** [comment]
Okay, stopped AWCC in the services.msc first, updated AlienFX to latest version (7.4.4.1) and changed colors again: 
[data.zip](https://github.com/T-Troll/alienfx-tools/files/9955272/data.zip)

**Comment by T-Troll** [comment]
To make sure - is it a log from AWCC or my GUI?
If first, did AWCC change colors now?

**Comment by T-Troll** [comment]
Ohshi... Seems like i found a source of this issue.
Please check this test version (both CLI and GUI inside) - [TestRelease.zip](https://github.com/T-Troll/alienfx-tools/files/9958026/TestRelease.zip)

It should work now.

**Comment by MaluNoPeleke** [comment]
https://photos.app.goo.gl/qu5UfAtV7gem2wCd7

**Comment by T-Troll** [comment]
Got it. Did the same happened in case you just press "Refresh" button in UI (not in Ambient mode)?

**Comment by T-Troll** [comment]
NB: I configure out you have the same light into the 2 groups involved into Ambient. So it set 2 times with different color, which can provide this effect. It's useless, in fact. I will provide check for double light set (not only Ambient one) into the next release.

**Comment by MaluNoPeleke** [comment]
Okay, great

**Comment by T-Troll** [comment]
But what about just set some color and press "Refresh" button? Did it provide other color flash as well?

**Comment by T-Troll** [comment]
Please check [7.4.6](https://github.com/T-Troll/alienfx-tools/releases/tag/7.4.6), it's having previous fixes for your device, as well as lights de-duplication and some other bug fixes.

**Comment by MaluNoPeleke** [comment]
Thanks, but it is still hard for me to figure out how to configure it correctly. Lights are flickering purple with ambient mode turned on 
![image](https://user-images.githubusercontent.com/1446541/201214944-bd3ed8ad-9b98-4634-93a0-6accdf5fbb5d.png)
If I start from scratch, what exactly should I click on? It is not easy to identify for example which areas of the screen/RGBs is currently selected and which not. Maybe you could improve the usability to make it more intuitive?

**Comment by T-Troll** [comment]
> Lights are flickering purple with ambient mode turned on

Did you check this?
> But what about just set some color and press "Refresh" button? Did it provide other color flash as well?

>  It is not easy to identify for example which areas of the screen/RGBs is currently selected and which not.

Yes, i don't so happy about current selection as well, but it's hard to do it more contrast. But i'll try.
The concept is easy - you have zone, and map it to follow the screen part color. Screen grid can be resized as well, so tune it to your light's position. I think you need 3x3 grid in your case.

**Comment by MaluNoPeleke** [comment]
Yes, checked it with single color mode as well (Effect dropdown on "Off").

**Comment by T-Troll** [comment]
Did you get the same effect (other color blink) then effect off and you press "Refresh" button?

**Comment by MaluNoPeleke** [comment]
Yes

**Comment by T-Troll** [comment]
Splendid!
Seems like i know the source of this... Let me build test version.

**Comment by T-Troll** [comment]
Ok, let's try this version - [alienfx-gui.zip](https://github.com/T-Troll/alienfx-tools/files/9994509/alienfx-gui.zip)
It should only blink violet at start, not every update.

In case it works correctly - great.
In case it not or don't switch colors at all - please do this:
- Make sure AWCCService is running (NOT stopped!)
- Open USBPcap, start recording
- Open CMD into tools folder and issue the command `alienfx-cli high-level setcolor=0,0,255,0,0 setcolor=0,1,0,255,0` (colors should be changed on monitor)
- Share the .pcap file.

This give me a hint how AWCC set separate lights via it's SDK, not AWCC in itself.

**Comment by translucentfocus** [comment]
> Ok, let's try this version - [alienfx-gui.zip](https://github.com/T-Troll/alienfx-tools/files/9994509/alienfx-gui.zip) It should only blink violet at start, not every update.
> 
> In case it works correctly - great. In case it not or don't switch colors at all - please do this:
> 
> * Make sure AWCCService is running (NOT stopped!)
> * Open USBPcap, start recording
> * Open CMD into tools folder and issue the command `alienfx-cli high-level set-one=0,0,255,0,0 set-one=0,1,0,255,0` (colors should be changed on monitor)
> * Share the .pcap file.
> 
> This give me a hint how AWCC set separate lights via it's SDK, not AWCC in itself.

I was able to follow your directions on our test sample.

[Ambient.txt](https://github.com/T-Troll/alienfx-tools/files/9994713/Ambient.txt)

**Comment by translucentfocus** [comment]
> It's a mystery. You have the same device ID as before, but it used a completely different interface to set lights! But, being honest, more simple one.
> 
> I don't know, is it a firmware or different hardware mode. Anyway, let me implement it.
There are at least two different firmware versions of the monitor that do change how things function.  It also appears that @MaluNoPeleke is in the EU, so there may be difference between the EU and NA (North America) versions of the monitor.

**Comment by translucentfocus** [comment]
> Thanks, but it is still hard for me to figure out how to configure it correctly. Lights are flickering purple with ambient mode turned on ![image](https://user-images.githubusercontent.com/1446541/201214944-bd3ed8ad-9b98-4634-93a0-6accdf5fbb5d.png) If I start from scratch, what exactly should I click on? It is not easy to identify for example which areas of the screen/RGBs is currently selected and which not. Maybe you could improve the usability to make it more intuitive?

This is my fault.  I can gie @T-Troll a better default config for this monitor.  For example, I made the mistake of calling it **Stand** light, when the firmware controls call it the **Ring** light.  In addition, there's no reason to make it a literal ring in the grid because it is seen as _one_ light.  This means I can simplify the grid substantially.

**Comment by T-Troll** [comment]
@translucentfocus Ooohh... welcome back! 

Being honest, this interface is a most problematic among i try to support.

> There are at least two different firmware versions of the monitor that do change how things function.

No, it was my fault - i think @MaluNoPeleke share AWCC log, but it was mine GUI log. I configure it out later and found a bug providing this result.

>  In addition, there's no reason to make it a literal ring in the grid

No, it has a reason. Making precise grid have some benefits for other effect types, especially grid ones. I'm working on it now, it will be much better soon.

Thank you for pcap, i'll check it. 

For now, seems like lights is mixed if set in series, so i need to find a way to reset/update previous set before starting next.

**Comment by T-Troll** [comment]
Hmm... If the log @translucentfocus present me from high-level cli - things become way more simple!

Can anyone capture another high-level cli log for me, but the command now will be different:
`alienfx-cli high-level set-one=0,0,255,0,0 set-one=0,1,0,255,0 set-one=0,0,0,0,255`?

Also, please, check - is light 0 blink violet on not into the process?

**Comment by translucentfocus** [comment]
> Hmm... If the log @translucentfocus present me from high-level cli - things become way more simple!
> 
> Can anyone capture another high-level cli log for me, but the command now will be different: `alienfx-cli high-level set-one=0,0,255,0,0 set-one=0,1,0,255,0 set-one=0,0,0,0,255`?
> 
> Also, please, check - is light 0 blink violet on not into the process?
[set-onetest2.txt](https://github.com/T-Troll/alienfx-tools/files/9997251/set-onetest2.txt)

Hopefully @MaluNoPeleke can also give you logs just to verify I'm not screwing something up 😆

I checked, and light 0 doesn't seem to do anything when I run
`alienfx-cli high-level set-one=0,0,255,0,0 set-one=0,1,0,255,0 set-one=0,0,0,0,255`

**Comment by MaluNoPeleke** [comment]
Here is a log of this and the previous command. The last one was setting a color via the GUI.
[light2.zip](https://github.com/T-Troll/alienfx-tools/files/9998134/light2.zip)

**Comment by T-Troll** [comment]
Guys, are you sure you send me log from last command (3 x set-one), not previous (2 x set-one)?

I see only 2 light sets (first and second set-one) into it - in both logs (@MaluNoPeleke, i even see you have AWCC service running ^_^, but seems like @translucentfocus  doesn't have it).

Anyway... Open issue is solved, one of the commands i use as "block start" is useless (or even harmful), so i already remove it from yesterday's 7.5.0. It can be the source of violet flash.

So... 7.5.0 should work better. But... There is a probability it can only change color once (not so big one, according to what AWCC do in @MaluNoPeleke example). Please check (you can also use latest CLI command for test, but without "high-level").

**Comment by MaluNoPeleke** [comment]
I tried out 7.5.0 and it works fine, even in Ambient mode. Thanks a lot for your efforts!
Do you know what the performance impact is in that mode?

**Comment by T-Troll** [comment]
> Do you know what the performance impact is in that mode?

Yes. 1.5-4% of 6-cores CPU depends on screen resolution, and it's grid size.
